addons/resource/models/resource.py

```diff
@@ -408,10 +408,10 @@ def _get_day_leave_intervals(self, day_date, start_time, end_time, resource_id):
             start_datetime=datetime.datetime.combine(day_date, start_time),
             end_datetime=datetime.datetime.combine(day_date, end_time))
 
-        final_intervals = filter(lambda i: i, [
-            self._interval_and(leave_interval, work_interval)
-            for leave_interval in leaves_intervals
-            for work_interval in working_intervals])
+        final_intervals = [i for i in
```

Comment:
In the second commit you have a mismatch of something done in the first commit and modified in the second one :) . I suppose it is due to separating commits in a working branch. Just ensure the first commit contains the fix and the second one the support of timezones.

Concerning the code itself, as I see you used both, a filter or a list comprehension are both fine to me.